### DISCUSSION TITLE
Averaging signal lever

### DISCUSSION THREAD

**Discussion by zygmund2000** [PE5PVB/TEF6686_ESP32#508] [Ideas]
Hi,

I would like to propose a change to the method of averaging the signal level, currently it is based on a low-pass filter formula:
SAvg = (((SAvg * 9) + 5) / 10) + SStatus;
While trying to experiment with this, I changed it to something simpler:
SAvg = (alpha * SStatus * 10) + (1 - alpha) * SAvg; where alpha=0.1
Unfortunately, the effect is still the same.
My idea is to reduce the value refresh, preferably using sampling rate conversion, this should reduce the flickering of decimal value readings with full 0.1 resolution.
In addition, it is worth changing the display method to Sprite, e.g. something like this in my code:
         SignalSprite.fillSprite(BackgroundColor);
         SignalSprite.loadFont(FONT48);
         SignalSprite.drawString(String(SStatus / 10.0 , 1) + " ", 92, 0);
         SignalSprite.unloadFont();
         SignalSprite.pushSprite(225, 108);

This look nice for me, also I rendered smaller monospace font for this - 36.

**Comment by PE5PVB** [comment]
Yes, I need to switch to sprite anyway to have multiple skin posibilities. I'm not sure what you want to change on the behaviour of averaging. Do you want to have a more stable .x readout?

You can also try to play with your smoothing parameter (alpha) to get a better result.

**Reply by sergey20178888** [reply]
Как с вами можно связаться. Я хочу в Tef6686 добавить интернет радио.  Я готов за это платить.

**Comment by zygmund2000** [comment]
Lower alpha slow down slope of signal rise or fall and it take too much time to see value but not solve unstable .x readout because incoming signal data are to fast.

**Comment by PE5PVB** [comment]
Maybe split it in two? So average the x. and .x seperatly?

**Comment by zygmund2000** [comment]
Maybe it could work, I still insist that it would be best to show every tenth value or something like that or convert it to lower rate.\
The time interval between samples looks much smaller than 100 ms and it would be better if it was larger eg. 500ms.

**Comment by PE5PVB** [comment]
Yes, but a jumping meter can be also very annoying. And the .x values aren't that accurate.

**Comment by zygmund2000** [comment]
Ok, so combine of two methods low pass and lower rate will look more professional
To be clear -  I'm not talking about state of art signal meter but only eliminate actual excessive speed of .x value changes.

**Comment by zygmund2000** [comment]
My workaround at the moment is like this:

```
  if (millis() >= SignalTimer + 100) {
    SignalTimer = millis();
      if (advancedRDS) {
        //tftReplace(1, String(SStatusold / 10.0 , 1) + " ", String(SStatusprint / 10.0 , 1) + " ", 277, 51, PrimaryColor, PrimaryColorSmooth, 16);
      } else {
        SignalSprite.fillSprite(BackgroundColor);
        SignalSprite.loadFont(FONT48);
        SignalSprite.drawString(String(SStatus / 10.0 , 1) + " ", 92, 0);
        SignalSprite.unloadFont();
        SignalSprite.pushSprite(225, 108);
    }
  }
```
And moved smeter bar outside that timer to work smoothly.
I am not satisfied with this solution, but I achieved the effect almost as I would like.

**Reply by PE5PVB** [reply]
If you set the decimal in the same font you can ditch the SignalSprite.unloadFont(); and move SignalSprite.loadFont(FONT48); to the setup function. This will increase the speed.
